Switch from fontForCaretSelection to returning the information in the stylingDictionary to remove more SPI.

megangardner opened the pull request:
#### ea623fe8675130f284b0f115793dcbbf53bf6987
<pre>
Switch from fontForCaretSelection to returning the information in the stylingDictionary to remove more SPI.
<a href="https://bugs.webkit.org/show_bug.cgi?id=265708">https://bugs.webkit.org/show_bug.cgi?id=265708</a>
<a href="https://rdar.apple.com/119057964">rdar://119057964</a>

Reviewed by Wenson Hsieh.

In our continued effort to move off of SPI, remove this call and instead return the information in a API path.

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView textStylingAtPosition:inDirection:]):
(-[WKContentView fontForCaretSelection]): Deleted.

Canonical link: <a href="https://commits.webkit.org/271515@main">https://commits.webkit.org/271515@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/21e3bedc1c60d6e147f3dc8ebdd499df83985a25

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28605 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7250 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/29987 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31245 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/26121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9363 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4620 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/26214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28875 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5992 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24599 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/5221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5362 "Passed tests") | | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/32611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26192 "Found 1 new API test failure: TestWebKitAPI.AutocorrectionTests.FontAtCaretWhenUsingUICTFontTextStyle (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26045 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31609 "layout-tests (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5324 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3496 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/29389 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6933 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6855 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5788 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5843 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -4237,22 +4237,15 @@ - (NSDictionary *)textStylingAtPosition:(UITextPosition *)position inDirection:(
-        fontDescriptor = adoptCF(CTFontDescriptorCreateCopyWithSymbolicTraits(fontDescriptor.get(), symbolicTraits, symbolicTraits));
     
-    RetainPtr<CTFontRef> font = adoptCF(CTFontCreateWithFontDescriptor(fontDescriptor.get(), 10, nullptr));
+    // Moved from old fontForCaretSelection implementation, removed because SPI
```
Nit - not sure this comment is necessary. If we want to keep this, then I'd suggest:
```suggestion
    // Setting a font here allows us to avoid implementing the private method `-fontForCaretSelection`.
```

megangardner committed:
Switch from fontForCaretSelection to returning the information in the stylingDictionary to remove more SPI.
https://bugs.webkit.org/show_bug.cgi?id=265708
rdar://119057964

Reviewed by Wenson Hsieh.

In our continued effort to move off of SPI, remove this call and instead return the information in a API path.

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView textStylingAtPosition:inDirection:]):
(-[WKContentView fontForCaretSelection]): Deleted.

Canonical link: https://commits.webkit.org/271515@main
